### PR TITLE
Issue 2026 total reaction

### LINKED ISSUE #2026: Check that all reactions are included in the "sum of interfacial current densities"
We could either manually update a list or check the subtrees somehow

### PR DESCRIPTION
**Pull request by tinosulzer**
# Description

Creates a separate submodel for the total kinetics ("sum of ...." variables
Continuing to break up #2073 

Fixes #2026 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

# Key checklist:

- [ ] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [ ] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [ ] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: pybamm/models/submodels/interface/kinetics/total_kinetics.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+# Total kinetics class, summing up contributions from all reactions
+#
+import pybamm
+
+
+class TotalKinetics(pybamm.BaseSubModel):
+    """
+    Total kinetics class, summing up contributions from all reactions
+
+    Parameters
```
Comment (rtimms) [inline pybamm/models/submodels/interface/kinetics/total_kinetics.py:11]: docstring needs updating here

File: pybamm/models/submodels/interface/kinetics/total_kinetics.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+                        f"{reaction_name}exchange ".capitalize()
+                        + "current density [A.m-2]": j0_dim,
+                        f"{reaction_name}exchange ".capitalize()
+                        + "current density per volume [A.m-3]": i_typ / L_x * j0,
+                    }
+                )
+
+            # Sum variables
+            if pybamm.xyz_average(j_p).id == pybamm.Scalar(0).id:
+                a_p = j_p  # zero
```
Comment (rtimms) [inline pybamm/models/submodels/interface/kinetics/total_kinetics.py:152]: can you add a more detailed comment here? on first read setting `a` to zero seems wrong

**Review by rtimms — APPROVED** [review]
thanks, @tinosulzer looks good! just a couple of minor comments

**Comment by rtimms** [comment]
Yeah that’s clearer, thanks!

File: pybamm/models/submodels/interface/kinetics/total_kinetics.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+                        s_n, "negative electrode", "current collector"
+                    ),
+                    zero_s,
+                    pybamm.FullBroadcast(
+                        s_p, "positive electrode", "current collector"
+                    ),
+                )
+
+            # Override print_name
+            j.print_name = "J"
```
Comment (brosaplanella) [inline pybamm/models/submodels/interface/kinetics/total_kinetics.py:188]: For the print names, do we need to add labels of which reaction they are?

**Review by brosaplanella — APPROVED** [review]
Looks good, thanks @tinosulzer!

File: pybamm/models/submodels/interface/kinetics/total_kinetics.py (hunk 3 above)
Comment (tinosulzer) [reply pybamm/models/submodels/interface/kinetics/total_kinetics.py:188]: probably, but that is a different issue

**Pull request merged**